src/test/ui/associated-types/associated-types-project-from-hrtb-in-struct.stderr

```diff
@@ -3,6 +3,8 @@ error[E0212]: cannot extract an associated type from a higher-ranked trait bound
    |
 LL |     field: I::A
    |            ^^^^
+   |
+   = note: this is not currently representable in Rust
```

Comment:
Hmm, this says "not representable", but you can still do things like:

```rust
pub trait Foo<T> {
    type A;
    fn get(&self, t: T) -> Self::A;
}

struct SomeStruct<'a, I : for<'x> Foo<&'x isize>> {
    field: <I as Foo<&'a isize>>::A
}
```

And it’ll work fine, likely doing exactly what you’d expect?

Reply:
I'll have a follow up PR with only the logic for this suggestion, as it will be more involved.